provider/datagen/src/transform/collator/transform.rs

```diff
@@ -0,0 +1,339 @@
+                            s.push_str(extension_str);
+                        }
+                    }
+                } else {
+                    // "standard" is the default for all but two languages: sv and zh.
+                    // Since there are only two special cases, hard-coding them
+                    // here for now instead of making the defaulting fancy and data driven.
+                    // The Swedish naming seems ad hoc from
+                    // https://unicode-org.atlassian.net/browse/CLDR-679 .
+
+                    if langid.language == "zh" {
+                        s.push_str("_pinyin");
+                    } else if langid.language == "sv" {
+                        s.push_str("_reformed");
+                    } else {
+                        s.push_str("_standard");
+                    }
+                }
+
+                let guard = self.data.read().unwrap();
+
+                if let Some($toml_data) = guard.as_ref().unwrap().get(&s) {
+                    Ok(DataResponse {
+                        metadata: DataResponseMetadata::default(),
+                        // The struct conversion is macro-based instead of
+                        // using a method on the Serde struct, because the
+                        // method approach caused lifetime issues that I
+                        // didn't know how to solve.
+                        payload: Some(DataPayload::from_owned($conversion)),
+                    })
+                } else {
+                    Err(DataErrorKind::MissingResourceOptions.with_req($marker::KEY, req))
+                }
+            }
+        }
+
+        icu_provider::impl_dyn_provider!(
+            $provider,
+            [$marker,],
+            SERDE_SE,
+            ITERABLE_SERDE_SE,
+            DATA_CONVERTER
+        );
+
+        impl IterableResourceProvider<$marker> for $provider {
+            fn supported_options(
+                &self,
+            ) -> Result<Box<dyn Iterator<Item = ResourceOptions>>, DataError> {
+                if self.data.read().unwrap().is_none() {
+                    let root_dir = self.source.get_coll_root()?;
+                    let data = $provider::load_data(root_dir).map_err(|e| {
+                        DataError::custom("Could not create provider").with_display_context(&e)
+                    })?;
+                    *self.data.write().unwrap() = Some(data);
+                }
+
+                let guard = self.data.read().unwrap();
+
+                let list: Vec<ResourceOptions> = guard
+                    .as_ref()
+                    .unwrap()
+                    .keys()
+                    .map(|k| {
+                        let (language, variant) = k.rsplit_once('_').unwrap();
+                        let langid = if language == "root" {
+                            LanguageIdentifier::default()
+                        } else {
+                            language.parse().unwrap()
```

Comment:
Nit: use `filter_map` to exclude ignore invalid data instead of panicking.

Reply:
Done even though I think it's unhelpful to silently ignore badly-named files instead of alerting the user of datagen.

Reply:
In that case return a `DataError`, but don't panic